src/lib/core/coercion/date-property.ts

```diff
@@ -0,0 +1,5 @@
+/** Coerces a data-bound value (typically a string) to a Date. */
+export function coerceDateProperty(value: any, fallbackValue = new Date()): Date {
+  let timestamp = Date.parse(value);
```

Comment:
I'm concerned about doing this date parsing here, since ultimately the parsing will be part of what we do for i18n. JavaScript's `Date.parse` is quite terrible; a real solution needs something like Closure `goog.date` or moment.js.